bundle/manifests/cryostat-operator.clusterserviceversion.yaml

```diff
@@ -509,7 +509,13 @@ spec:
                     port: 8081
                   initialDelaySeconds: 5
                   periodSeconds: 10
-                resources: {}
+                resources:
+                  limits:
+                    cpu: 500m
+                    memory: 123Mi
```

Comment:
123? Kind of an odd amount. 128?

If you actually run the Operator and use `oc describe` on its Pod, you should be able to get some values for the amount of resources that our Operator actually uses in practice. Have you checked that yet?

Reply:
> 123? Kind of an odd amount. 128?

Weird :D Guess I typed 3 and mistook it for 8 since its pretty much "half" of 8 somehow...

> If you actually run the Operator and use oc describe on its Pod, you should be able to get some values for the amount of resources that our Operator actually uses in practice. Have you checked that yet?

Right, I run it and it somehow did not show any resource usage. I got the logs [here](https://gist.github.com/tthvo/d7ffeedefa3245e397e2eecb30360b6b).

I also tried running `oc adm top pod cryostat-operator-controller-manager-8588cb98b8-zrgwl` but give:
```
error: Metrics API not available
```
I also try seeing the deployment configuration for the operator. It seems to not include any cpu or memory requests/limits. Maybe I didnt understand it correctly or somehow its got removed (already run `make bundle` after changes)?

```
Name:                   cryostat-operator-controller-manager
Namespace:              default
CreationTimestamp:      Tue, 05 Jul 2022 12:26:14 -0400
Labels:                 olm.deployment-spec-hash=576685dcf4
                        olm.owner=cryostat-operator.v2.2.0-dev
                        olm.owner.kind=ClusterServiceVersion
                        olm.owner.namespace=default
                        operators.coreos.com/cryostat-operator.default=
Annotations:            deployment.kubernetes.io/revision: 1
Selector:               control-plane=controller-manager
Replicas:               1 desired | 1 updated | 1 total | 1 available | 0 unavailable
StrategyType:           RollingUpdate
MinReadySeconds:        0
RollingUpdateStrategy:  25% max unavailable, 25% max surge
Pod Template:
  Labels:           control-plane=controller-manager
  Annotations:      alm-examples:
                      [
                        {
                          "apiVersion": "operator.cryostat.io/v1beta1",
                          "kind": "Cryostat",
                          "metadata": {
                            "name": "cryostat-sample"
                          },
                          "spec": {
                            "enableCertManager": true,
                            "eventTemplates": [],
                            "minimal": false,
                            "reportOptions": {
                              "replicas": 0
                            },
                            "storageOptions": {
                              "pvc": {
                                "annotations": {},
                                "labels": {},
                                "spec": {}
                              }
                            },
                            "trustedCertSecrets": []
                          }
                        },
                        {
                          "apiVersion": "operator.cryostat.io/v1beta1",
                          "kind": "FlightRecorder",
                          "metadata": {
                            "name": "example-flightrecorder"
                          },
                          "spec": {
                            "recordingSelector": {
                              "matchLabels": {
                                "operator.cryostat.io/flightrecorder": "example-flightrecorder"
                              }
                            }
                          }
                        },
                        {
                          "apiVersion": "operator.cryostat.io/v1beta1",
                          "kind": "Recording",
                          "metadata": {
                            "name": "example-recording"
                          },
                          "spec": {
                            "archive": true,
                            "duration": "30s",
                            "eventOptions": [
                              "template=ALL"
                            ],
                            "flightRecorder": {
                              "name": "example-flightrecorder"
                            },
                            "name": "example-recording"
                          }
                        }
                      ]
                    capabilities: Basic Install
                    categories: Monitoring, Developer Tools
                    containerImage: quay.io/cryostat/cryostat-operator:2.2.0-dev
                    createdAt: 2021-04-30 00:00:00
                    description: JVM monitoring and profiling tool
                    olm.operatorGroup: operator-sdk-og
                    olm.operatorNamespace: default
                    olm.targetNamespaces: default
                    operatorframework.io/initialization-resource:
                      {
                        "apiVersion": "operator.cryostat.io/v1beta1",
                        "kind": "Cryostat",
                        "metadata": {
                          "name": "cryostat-sample"
                        },
                        "spec": {
                          "enableCertManager": true,
                          "minimal": false,
                          "reportOptions": {
                            "replicas": 0
                          }
                        }
                      }
                    operatorframework.io/properties:
                      {"properties":[{"type":"olm.package","value":{"packageName":"cryostat-operator","version":"2.2.0-dev"}},{"type":"olm.gvk","value":{"group"...
                    operators.operatorframework.io/builder: operator-sdk-v1.4.0+git
                    operators.operatorframework.io/internal-objects: ["flightrecorders.operator.cryostat.io","recordings.operator.cryostat.io"]
                    operators.operatorframework.io/project_layout: go.kubebuilder.io/v3
                    repository: github.com/cryostatio/cryostat-operator
                    support: Cryostat Community
  Service Account:  cryostat-operator-service-account
  Containers:
   manager:
    Image:      quay.io/thvo/cryostat-operator:2.2.0-dev
    Port:       <none>
    Host Port:  <none>
    Command:
      /manager
    Args:
      --leader-elect
    Liveness:   http-get http://:8081/healthz delay=15s timeout=1s period=20s #success=1 #failure=3
    Readiness:  http-get http://:8081/readyz delay=5s timeout=1s period=10s #success=1 #failure=3
    Environment:
      RELATED_IMAGE_CORE:        quay.io/cryostat/cryostat:latest
      RELATED_IMAGE_DATASOURCE:  quay.io/cryostat/jfr-datasource:latest
      RELATED_IMAGE_GRAFANA:     quay.io/cryostat/cryostat-grafana-dashboard:latest
      RELATED_IMAGE_REPORTS:     quay.io/cryostat/cryostat-reports:latest
      WATCH_NAMESPACE:            (v1:metadata.annotations['olm.targetNamespaces'])
      OPERATOR_CONDITION_NAME:   cryostat-operator.v2.2.0-dev
    Mounts:                      <none>
  Volumes:                       <none>
Conditions:
  Type           Status  Reason
  ----           ------  ------
  Available      True    MinimumReplicasAvailable
  Progressing    True    NewReplicaSetAvailable
OldReplicaSets:  <none>
NewReplicaSet:   cryostat-operator-controller-manager-8588cb98b8 (1/1 replicas created)
Events:
  Type    Reason             Age   From                   Message
  ----    ------             ----  ----                   -------
  Normal  ScalingReplicaSet  14m   deployment-controller  Scaled up replica set cryostat-operator-controller-manager-8588cb98b8 to 1
```

Reply:
> I also tried running oc adm top pod cryostat-operator-controller-manager-8588cb98b8-zrgwl but give:

Look like I need to have [metrics-server](https://github.com/kubernetes-sigs/metrics-server) service. I will try this.

Reply:
I think this is the one?
```
NAME                                                    CPU(cores)   MEMORY(bytes)   
cryostat-operator-controller-manager-8588cb98b8-5psst   1m           16Mi
```

Reply:
I am a bit unsure why we set resource request to 64Mi memory and 10m but the operator uses less than that? Was the resource request not applied at all?

Reply:
Any guidance on this @andrewazores @ebaron?

Reply:
We want to ensure we don't set the values too low and then the operator is resource starved or killed.

Reply:
Going by the limited info we have so far then, I would suggest these as defaults:

```
requests:
  cpu: 100m
  memory: 64Mi
limits:
  cpu: 1000m
  memory: 256Mi
```

The work done shouldn't be very CPU intensive, in fact I would expect it's mostly I/O bound given the amount of k8s API communication it's doing, so 1/10th of a CPU core is probably enough most of the time. I don't think it's doing anything in parallel so more than 1 full core is unnecessary. 64Mi of memory is double the observed steady state so should be plenty including allowance for garbage to accumulate and be collected during active periods, while also being a pretty small request amount in the grand scheme of things. If memory usage gets past 4x that already generous amount then something is probably wrong and the container should probably be restarted.

Reply:
Seems reasonable

Reply:
Ahh understood! Applied changes :D

Reply:
>  I wonder if oc adm top pod is behaving like the standard top utility and just printing what's currently in use, not the requests/limits. That would explain why the CPU usage is at 1 millicore, which is effectively zero. I doubt that's the amount assigned, it's just the measured amount at the time you ran the command. If you weren't modifying the Cryostat CR at the time or something then the Operator controller manager was most likely just sitting there idling, waiting for operator-sdk to notify it of some work to do and send it through the reconciliation loop.

I found:
```
$ oc adm top
Show usage statistics of resources on the server

 This command analyzes resources managed by the platform and presents current usage statistics.
Usage:
  oc adm top [flags]

Available Commands:
  images       Show usage statistics for images
  imagestreams Show usage statistics for image streams
  node         Display resource (CPU/memory) usage of nodes
  pod          Display resource (CPU/memory) usage of pods

Use "oc adm <command> --help" for more information about a given command.
Use "oc options" for a list of global command-line options (applies to all commands).
```

Guess I was a bit confused on what it means by `current` :D